_posts/2019-07-27-GettingStartedWithProtobufGrpc.md

```diff
@@ -0,0 +1,306 @@
+---
+layout: post
+title: Getting started with gRPC. Part 1
+categories: gRPC
+tags: [Protobuf, gRPC, HTTP/2]
+author: PrakharSrivastav
+---
+# Motivation and target audience
+
+There are several blog posts on the internet touching one or several shiny features that gRPC framework boasts about. There is also the [official documentation](https://grpc.io/docs/) from gRPC. It is very easy to get lost in all the jargons, terminology and the code examples. This ends up making readers confused and frustrated with a very limited understanding of what gRPC offers.
+
+The motivation behind writing this blog post is to gently introduce the gRPC framework to interested developers and the architects, skipping the more advanced details. This blog starts with setting up an initial context about the gRPC framework, followed by discussing a few of its promising features. We then highlight typical applications that can be build using gRPC and introduce the readers to a simple workflow for a small to medium scale project. Wherever possible, we have provided references to more detailed sources.
+
+This blog focusses only on introducing gRPC as a concept and provide a convenient starting point. If you want to get more detailed instructions, the best place would be the [README.md](https://github.com/sysco-middleware/workshop-grpc/blob/master/README.md) file in the Github repository specified in the next section.
```

Comment:
What if we make this post standalone, and at the end point to another post that introduces the gRPC workshop? I'd also prefer to have an standalone repo for samples from this post detached from workshop repo to make it easier to start as a reader (I know this could cost some boilerplate between repos, but at the expense of be easier to start with)

Reply:
Made a fix for this in the current commit. This blog has a standalone repo https://github.com/PrakharSrivastav/grpc-protobuf-starter, which mimics the same example. In the end, I have pointed to the workshop repo for a more detailed example.